scipy/sparse/csgraph/_shortest_path.pyx

```diff
@@ -166,12 +166,20 @@ def shortest_path(csgraph, method='auto',
         issparse = isspmatrix(csgraph)
         if issparse:
             Nk = csgraph.nnz
+            if isspmatrix_csr(csgraph) or isspmatrix_csc(csgraph):
```

Comment:
Since we already know it's a sparse matrix, we can do:

```python
if csgraph.format in ('csr', 'csc', 'coo'):
    edges = csgraph.data
else:
    edges = csgraph.tocoo().data
```

(I suggest using `tocoo()` because the data array is equivalent and converting to COO is often faster.)